### ISSUE TITLE
Bug: id-blacklist detects an identifier using `number` but id-denylist does not

### ISSUE THREAD

**Issue by Stargator** [eslint/eslint#15188]
### Environment

Node version: v10.24.1
npm version: npm v7
Local ESLint version: 7.32
Global ESLint version: N/A
Operating System: MacOS

The project is written in Angular v11.

When I have the below defined rules for `files: *.ts`, the id-denylist does not detect an identifier using `number`, but if I turn id-blacklist on (set the value from `off` to `error` the id-blacklist will detect the misnamed identifier.

It is my understanding that id-blacklist is deprecated and replaced by id-denylist, in this scenario I seem to still need to use id-blacklist in order to catch the lint issues I want resolved.

I know id-denylist **does not** cover object properties (so you can still use objects you do not have control over), but I don't think these apply. One issue is in an interface I defined and the others are in variables I defined.

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```yaml
  - files:
      - '*.ts'
    parserOptions:
      createDefaultProgram: true
      ecmaVersion: 2020
      project:
        - tsconfig.json
        - e2e/tsconfig.json
      sourceType: module
    extends:
      - 'eslint:recommended'
      - 'plugin:@angular-eslint/recommended'
      - 'plugin:@angular-eslint/ng-cli-compat'
      - 'plugin:@angular-eslint/ng-cli-compat--formatting-add-on'
      - 'plugin:@angular-eslint/template/process-inline-templates'
      - 'plugin:@typescript-eslint/recommended'
      - 'plugin:import/recommended'
      - 'plugin:import/typescript'
      - 'plugin:promise/recommended'
      - 'plugin:sonarjs/recommended'
    plugins:
      - '@angular-eslint'
      - '@typescript-eslint'
      - import
      - promise
      - sonarjs
    rules:
      '@angular-eslint/component-selector':
        - error
        - type: element
          prefix: retf
          style: kebab-case
      '@angular-eslint/contextual-decorator': error
      '@angular-eslint/directive-selector':
        - error
        - type: attribute
          prefix: retf
          style: camelCase
      '@angular-eslint/use-injectable-provided-in': error
      '@typescript-eslint/ban-ts-comment': off  # Enabled in recommended
      '@typescript-eslint/ban-types':
        - error
        - extendDefaults: true
          types:
            '{}': false
      '@typescript-eslint/explicit-member-accessibility':
        - error
        - accessibility: no-public
          overrides:
            parameterProperties: explicit
      '@typescript-eslint/naming-convention':
        - error
        - selector: objectLiteralProperty
          format: null
          leadingUnderscore: allowSingleOrDouble
      '@typescript-eslint/no-empty-function':
        - error
        - allow:
            - constructors
      '@typescript-eslint/no-explicit-any': off # set to warn
      '@typescript-eslint/no-namespace': off
      '@typescript-eslint/no-non-null-assertion': off # Enabled in recommended
      '@typescript-eslint/no-this-alias': off # Enabled in recommended
      '@typescript-eslint/no-unused-expressions': off # Enabled in recommended
      '@typescript-eslint/no-unused-vars': error
      '@typescript-eslint/quotes':
        - error
        - single
        - avoidEscape: true
          allowTemplateLiterals: true
      arrow-body-style:
        - error
        - as-needed
      arrow-parens:
        - error
        - as-needed
        - requireForBlockBody: true
      brace-style: error
      comma-dangle:
        - error
        - arrays: always
          exports: never
          functions: ignore
          imports: ignore
          objects: always
      complexity: error
      consistent-return: error
      curly: error
      dot-notation: error
      eqeqeq: error
      id-blacklist: off
      id-denylist:
        - error
        - any
          Number
          number
          String
          string
          Boolean
          boolean
          Undefined
          undefined
          Null
          null
      import/no-extraneous-dependencies: off
      import/no-internal-modules: off
      import/order:
        - error
        - groups:
            - builtin
            - external
            - internal
            - parent
            - sibling
            - index
            - object
          newlines-between: always
          alphabetize:
            order: asc
            caseInsensitive: false
          warnOnUnassignedImports: true
      indent:
        - error
        - 2
        - ArrayExpression: first
          CallExpression:
            arguments: first
          flatTernaryExpressions: true
          FunctionDeclaration:
            body: 1
            parameters: first
          FunctionExpression:
            body: 1
            parameters: first
          ImportDeclaration: first
          MemberExpression: 1
          ObjectExpression: first
          offsetTernaryExpressions: true
          outerIIFEBody: off
          SwitchCase: 1
          VariableDeclarator: first
      max-len:
        - error
        - code: 180
      no-await-in-loop: error
      no-console:
        - error
        - allow:
            - warn
            - error
      no-empty-function:
        - error
        - allow:
            - constructors
      no-empty-pattern: error
      no-extra-parens: error
      no-loss-of-precision: error
      no-multi-spaces: error
      no-multiple-empty-lines: error
      no-param-reassign: error
      no-promise-executor-return: error
      no-return-assign: error
      no-return-await: error
      no-shadow: error
      no-template-curly-in-string: error
      no-underscore-dangle:
        - error
        - allow:
            - __metadata
            - _latlng
            - _resetView
          allowAfterSuper: false
          allowAfterThis: false
          allowAfterThisConstructor: false
          allowFunctionParams: false
          enforceInMethodNames: false
      no-unreachable-loop: error
      no-unsafe-optional-chaining: error
      no-unused-expressions: off # Duplicated in @typescript-eslint
      no-use-before-define: off # Temporary, previously set to: off, but I'm interested in testing the impact of using it.
      no-useless-backreference: error
      no-useless-return: error
      promise/no-nesting: off # Enabled in recommended
      require-atomic-updates: error
      semi: off # Duplicated in @typescript-eslint
      semi-style:
        - error
        - last
      sonarjs/cognitive-complexity: off # Temporary, previously set to: warn
      quotes: off # Duplicated in @typescript-eslint
```

</details>

```typescript
export interface SheetColumnTypeIndexMap {
    latitude?: number[];
    longitude?: number[];
    text?: number[];
    number?: number[];
    address?: number[];
```

```typescript
export const sheet1 = {
  columnTypeIndexMap: {latitude: [3,], longitude: [4,], address: [],},
};
```

```typescript
    const sheetColumnTypeIndexMap: SheetColumnTypeIndexMap = {
      latitude: [],
      longitude: [],
      text: [],
      number: [],
      address: [],
    };
``````

### What did you expect to happen?

Lint output **from id-denylist ONLY**
Linting "angular-proj"...

/Users/user/Workspace/angular-proj/src/app/models/sheet-object.ts
  7:5  error  Identifier 'number' is restricted  id-denylist 

/Users/user/Workspace/angular-proj/src/app/services/upload-excel-file.service.ts
  107:7  error  Identifier 'number' is restricted  id-denylist

/Users/user/Workspace/angular-proj/src/assets/mock-data/sheets.ts
   61:67  error  Identifier 'number' is restricted  id-denylist

✖ 3 problems (3 errors, 0 warnings)

### What actually happened?

Lint output:
Linting "angular-proj"...

/Users/user/Workspace/angular-proj/src/app/models/sheet-object.ts
  7:5  error  Identifier 'number' is restricted  id-blacklist

/Users/user/Workspace/angular-proj/src/app/services/upload-excel-file.service.ts
  107:7  error  Identifier 'number' is restricted  id-blacklist

/Users/user/Workspace/angular-proj/src/assets/mock-data/sheets.ts
   61:67  error  Identifier 'number' is restricted  id-blacklist

✖ 3 problems (3 errors, 0 warnings)

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

While, sure I am willing to submit a PR, I have never contributed before so I don't know how to resolve this.

**Comment by Stargator** [comment]
@mdjermanovic Sure, I think I tried something like that before. But yeah...give me a moment...

Well...that worked. I was sure I had an error doing it that way. Oh well. Thank you @mdjermanovic!
